### PR TITLE
MINOR: Remove deprecated KafkaStreams constructors in docs

### PR DESCRIPTION
**Pull request by guozhangwang**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by guozhangwang** [comment]
@bbejeck @vvcephei

**Comment by mjsax** [comment]
Tutorial also needs an update. For example:
```
one can also construct a <code>StreamsConfig</code> object from the <code>props</code> map and then pass that object to the constructor,
```

**Comment by mjsax** [comment]
Also `config-streams.html`:
```
Construct a StreamsConfig instance from the Properties instance. For example:
```

**Comment by mjsax** [comment]
datatypes.html: `By setting default SerDes via a StreamsConfig instance.`

**Comment by mjsax** [comment]
There is two more in `dsl-api.html`  ("Writing to Kafka" section)

**Comment by mjsax** [comment]
Also multiple in "Interactive Queries" page

**Comment by mjsax** [comment]
Multiple ones in memory management: `Properties streamsConfiguration` -> `Properties props`

**Comment by mjsax** [comment]
Processor API section: `You can override this setting by specifying StreamsConfig.WINDOW_STORE_CHANGE_LOG_ADDITIONAL_RETENTION_MS_CONFIG in the StreamsConfig`

**Comment by mjsax** [comment]
security.html also uses `StreamsConfig` multiple times

**Comment by mjsax** [comment]
Maybe update `testing.html` from `Properties config` -> `Properties props` ?

**Comment by mjsax** [comment]
And "Writing a Streams Application": `The second argument is an instance of StreamsConfig, which defines the configuration for this specific topology.`

**Comment by mjsax** [comment]
ProcessorContext.java has it in its JavaDoc, too: `{@link org.apache.kafka.streams.StreamsConfig#StreamsConfig(Map) StreamsConfig(Map)}).`

**Comment by mjsax** [comment]
That should be all, if I did not miss any... Please double check. :)

**Comment by guozhangwang** [comment]
@mjsax I made a pass over all docs file and I think I should have got all of them.

**Review by mjsax — APPROVED** [review]
Thanks @guozhangwang! With all the HTML markup, it's hard to review... Hope I didn't miss anything...

LGTM.

File: docs/streams/tutorial.html [hunk 1]
```diff
@@ -208,9 +208,7 @@ <h4><a id="tutorial_code_pipe" href="#tutorial_code_pipe">Writing a first Stream
     <p>
         Note that we can always describe the topology as we did above at any given point while we are building it in the code, so as a user you can interactively "try and taste" your computational logic defined in the topology until you are happy with it.
         Suppose we are already done with this simple topology that just pipes data from one Kafka topic to another in an endless streaming manner,
-        we can now construct the Streams client with the two components we have just constructed above: the configuration map and the topology object
-        (one can also construct a <code>StreamsConfig</code> object from the <code>props</code> map and then pass that object to the constructor,
-        <code>KafkaStreams</code> have overloaded constructor functions to takes either type).
+        we can now construct the Streams client with the two components we have just constructed above: the configuration map and the topology object.
```
Comment (bbejeck) [inline docs/streams/tutorial.html]: super nit: `configuration map` -> `java.util.Properties` instance?

**Review by bbejeck — APPROVED** [review]
Thanks, @guozhangwang just one super-nit. LGTM

**Pull request merged**